src/test/java/org/jenkinsci/plugins/workflow/steps/scm/SCMStepTest.java

```diff
@@ -205,7 +205,7 @@ public InvalidChangelogSCM() { }
         @DataBoundConstructor
         public FakeChangeLogSCM() { }
         @Override public void checkout(Run<?,?> build, Launcher launcher, FilePath workspace, TaskListener listener, File changelogFile, SCMRevisionState baseline) throws IOException, InterruptedException {
-            // Unlike the superclass, which adds an XML header to the file, we just ignore the file.
+            // TODO: Need to find the appropriate thing to do here. Could be nothing.
```

Comment:
Right now, `InvalidChangelogSCM` and `FakeChangeLogSCM` both do the same exact thing. That's probably not the right thing. I don't know what the right thing is yet, so, I'm going to go figure that out.

If they really both just need to ignore the file, which is [what `InvalidChangelogSCM does](FakeChangeLogSCM), then maybe these classes can be combined into one. That's counter to the goal here, though, of having a test that specifically uses FakeChangeLogSCM

Reply:
I would leave `InvalidChangelogSCM` the way it was, since that tests the specific bugs with the Repo SCM and one other SCM that I can't remember. You shouldn't need to define `FakeChangelogSCM`, you can just import it from `jenkins-test-harness` and use it as-is.